### PR TITLE
[FIX] tool categories & additional checks.

### PR DESCRIPTION
**Pull request by oliveralka**
# Description

Fix for #5014
Fixes the tool categories for some tools. 
Adds functionality to check for duplicated entries (TOPP, Utils) 

Let me know if that is what you had in mind. 

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

**Review by timosachsenberg — APPROVED** [review]

**Comment by timosachsenberg** [comment]
nice. thanks!

**Comment by jpfeuffer** [comment]
One thing missing would be a warning when running the tool (i.e. a check in TOPPBase).
Yes, it introduced a dependency on Toolhandler (if not already present) but might be worth it. Often developers do not build the documentation and then it just is found in the nightlies.

**Comment by jpfeuffer** [comment]
https://github.com/OpenMS/OpenMS/blob/591cc3f1f130374b5aca12f38cd4c7b1fbf65508/src/openms/source/APPLICATIONS/TOPPBase.cpp#L39
We have a dependency already.

**Comment by timosachsenberg** [comment]
ah ok you mean in addition write out a warning to OPENMS_LOG_WARN ?

**Comment by jpfeuffer** [comment]
Maybe add a check for presence in either of the lists.

**Comment by timosachsenberg** [comment]
but this check is already in there
https://github.com/OpenMS/OpenMS/pull/5032/files#diff-38dd76e7d29ade2c6a692b72f0ba8e04a92b6f63d118a6f930a9f1088727e347R142
https://github.com/OpenMS/OpenMS/pull/5032/files#diff-38dd76e7d29ade2c6a692b72f0ba8e04a92b6f63d118a6f930a9f1088727e347R148

**Comment by cbielow** [comment]
I'd personally make this check a hard error.
Maybe this can even be done as `constexpr` compile warning once we are in C++17 land

**Comment by oliveralka** [comment]
Ok, so we could throw an exception (e.g. InvalidValue), instead of the `writeLog_`?

**Comment by cbielow** [comment]
yes

**Comment by jpfeuffer** [comment]
? sorry I overlooked that completely.

**Comment by oliveralka** [comment]
Since the "Tools/Utils" used in the TOPPBase_test are not registered in the ToolHandler - the tests fail. 

```
Message: the value 'TOPPBaseTest' was used but is not valid; If 'TOPPBaseTest' is a Util, add it to the util list in ToolHandler. If it is not, set the 'official' flag of the TOPPBase constructor to true.
515:  failed
```

I am actually not sure what to do here? Of course, we could add them to the ToolHandler, but they are not actually tools, or?

**Comment by timosachsenberg** [comment]
hmm the easiest solution might be adding a protected member: 
bool disable_toolhandler_test_ = false; 
to TOPPBase and set it in these tests to true (which you can query and skip the if / throwing the exception).

**Comment by timosachsenberg** [comment]
I would not add it to ToolHandler as we query those e.g., to get the list of executables etc.

File: src/openms/source/APPLICATIONS/TOPPBase.cpp [hunk 1]
```diff
@@ -132,34 +133,37 @@ namespace OpenMS
       verboseVersion_ += String(", Revision: ") + VersionInfo::getRevision() + "";
     }
 
-    // check if tool entries are in Utils and TOPP (duplication)
-    if (ToolHandler::checkDuplicated(tool_name_))
+    if (toolhandler_test_)
```
Comment (timosachsenberg) [inline src/openms/source/APPLICATIONS/TOPPBase.cpp:136]: nice but please add a comment why this can be disabled

**Review by timosachsenberg — APPROVED** [review]

**Comment by timosachsenberg** [comment]
http://cdash.openms.de/viewBuildError.php?buildid=67203

**Pull request merged**